### PR TITLE
Find files based on the study doc's enclosing directory

### PR DESCRIPTION
**Pull request by hepcat72**
## Summary Change Description

When loading a study doc, look for mzXML and peak annotation files in the directory containing the study doc by default.

While I was working on updating the contrib doc for version 3, I noted that the old commands did not have to change directories into the study directory in order to load it, and I consulted the old loading code and saw that it basically used the directory of the loading.yaml file.  I applied that strategy to the new loader.

Details:

- Fixed little documentation and f-string issues I encountered.
- Renamed a variable in the study loader: loader_class to "derived_class" for accuracy.
- Added a filename return to PeakAnnotationFilesLoader.get_file_and_format when the file is not found.

## Affected Issues/Pull Requests

- Resolves no documented issue
- Merges into main

## Review Notes

See comments in-line.

## Checklist

This pull request will be merged once the following requirements are met.  The
author and/or reviewers should uncheck any unmet requirements:

- Review requirements
  - Minimum approvals: 1 
  - No changes requested
  - All blocking issues resolved by reviewers
  - Specific reviewers: @__add_username_here__
    
  - Review period: 2 days 
- Associated issue/pull request requirements:
  
  - [x] All requirements in affected issues marked "resolved" are satisfied
  - [x] All required pull requests are merged *(or none)*
- Basic requirements
  
  - [x] [All linters pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#linting)
  - [x] [All tests pass](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#quality-control)
  - [x] All conflicts resolved
- Overhead requirements
  
  - [x] [New/changed method tests implemented/updated *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#test-implementation)
  - [ ] [Updated *Unreleased* section of `changelog.md` *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/changelog.md)
  - [x] [Migrations created & committed *(or no change)*](https://github.com/Princeton-LSI-ResearchComputing/tracebase/blob/main/CONTRIBUTING.md#migration-process)

### CODE REVIEW

File: DataRepo/data/tests/small_obob_mzxmls/small_obob_maven_6eaas_inf_glucose_mzxmls/BAT-xz971.mzXML
Comment (hepcat72) [inline DataRepo/data/tests/small_obob_mzxmls/small_obob_maven_6eaas_inf_glucose_mzxmls/BAT-xz971.mzXML:1]: C. I had to move the test mzXML files so they wouldn't get autoloaded, now that it defaults to the study directory.  Those files were never intended to be loaded together - they were for separate tests.

File: DataRepo/loaders/peak_annotation_files_loader.py [hunk 1]
```diff
@@ -216,7 +216,10 @@ def load_data(self):
         Args:
             None
         Exceptions:
-            None
+            Raises:
+                AggregatedErrorsSet
+            Buffers:
+                None
```
Comment (hepcat72) [inline DataRepo/loaders/peak_annotation_files_loader.py:222]: C. I noticed this docstring was not up to date.

File: DataRepo/loaders/study_loader.py [hunk 2]
```diff
@@ -223,7 +224,7 @@ def version_number(self) -> str:
         ERRORS=None,
     )
 
-    # NOTE: The instance copies this and adds to the copy
+    # NOTE: The constructor copies this and adds to it
```
Comment (hepcat72) [inline DataRepo/loaders/study_loader.py:227]: C. Clarity changes.

File: DataRepo/loaders/study_loader.py [hunk 3]
```diff
@@ -344,8 +352,8 @@ def __init__(self, *args, **kwargs):
         self.load_statuses = MultiLoadStatus()
 
         self.derived_loaders = {}
-        for loader_class in StudyLoader.__subclasses__():
-            self.derived_loaders[loader_class.__name__] = loader_class
+        for derived_class in StudyLoader.__subclasses__():
+            self.derived_loaders[derived_class.__name__] = derived_class
```
Comment (hepcat72) [inline DataRepo/loaders/study_loader.py:356]: C. Renamed for accuracy, and to distinguish the contained versus derived classes.

File: DataRepo/loaders/study_loader.py [hunk 4]
```diff
@@ -323,7 +325,13 @@ def __init__(self, *args, **kwargs):
         self.df_dict = None
 
         self.annot_files_dict = kwargs.pop("annot_files_dict", {})
-        mzxml_dir = kwargs.pop("mzxml_dir", None)
+
+        # Before the superclass constructor is called, we want to use the file path to get its enclosing directory as a
+        # default for the (custom derived class argument:) mzxml_dir
+        study_file = kwargs.get("file")
+        study_dir = None if study_file is None else os.path.dirname(study_file)
+        mzxml_dir = kwargs.pop("mzxml_dir", study_dir)
```
Comment (hepcat72) [inline DataRepo/loaders/study_loader.py:333]: C. This is the crux of the change (part 2).

**Review by narzouni — APPROVED** [review]
LGTM. I don't think I have any suggestions for code improvements on this PR.

**Pull request merged**